src/main/java/org/apache/zeppelin/interpreter/SnappyDataZeppelinInterpreter.java

```diff
@@ -725,93 +733,119 @@ public InterpreterResult interpret(String[] lines, InterpreterContext context) {
     }
   }
 
+  private void addDeployedJars() {
+    if (snappyGlobalLoader != null) {
+      URL[] allUrls = snappyGlobalLoader.getURLs();
+      if (addedURLS == null || (addedURLS.length != allUrls.length)) {
+        List<URL> added = new ArrayList<>();
+        for (URL url : allUrls) {
+          try {
+            sc.addJar(url.getFile());
+            added.add(url);
+          } catch (Throwable t) {
```

Comment:
What exceptions are expected? Better to catch only those?

Reply:
If a jar is already added it might throw exception. Ok will see what exactly that exception is.